src/test/java/com/codeus/winter/config/DefaultBeanFactoryTest.java

```diff
@@ -0,0 +1,203 @@
+     * Method under test: {@link DefaultBeanFactory#getBean(Class)}
+     */
+    @Test
+    @DisplayName("Test getBean(Class) with 'requiredType'; " +
+                 "when 'java.lang.Object'; " +
+                 "then throw BeanNotFoundException")
+    void testGetBeanWithRequiredTypeWhenJavaLangObjectThenThrowBeanNotFoundException() throws BeanNotFoundException {
+        Class<Object> requiredType = Object.class;
+
+        assertThrows(BeanNotFoundException.class, () -> beanFactory.getBean(requiredType));
+    }
+
+    /**
+     * Test {@link DefaultBeanFactory#getBean(Class)} with {@code requiredType}.
+     * <ul>
+     *   <li>When {@code java.lang.Object}.</li>
+     *   <li>Then throw {@link BeanNotFoundException}.</li>
+     * </ul>
+     * <p>
+     * Method under test: {@link DefaultBeanFactory#getBean(Class)}
+     */
+    @Test
+    @DisplayName("Test getBean(Class) with 'requiredType'")
+    void testGetBeanWithRequiredType() throws BeanNotFoundException, InvocationTargetException,
+            InstantiationException, IllegalAccessException, NoSuchMethodException {
+        var requiredType = Object.class;
+        var expected = beanFactory.createBean(requiredType);
+        var actual = beanFactory.getBean(requiredType);
+
+        assertEquals(expected, actual);
+    }
+
+    /**
+     * Test {@link DefaultBeanFactory#createBean(Class)} with {@code beanClass}.
+     * <ul>
+     *   <li>When {@code java.lang.Object}.</li>
+     * </ul>
+     * <p>
+     * Method under test: {@link DefaultBeanFactory#createBean(Class)}
+     */
+    @Test
+    @DisplayName("Test createBean(Class) with 'beanClass'; when 'java.lang.Object'")
+    void testCreateBeanWithBeanClassWhenJavaLangObject() throws NotUniqueBeanDefinitionException,
+            IllegalAccessException, InstantiationException, NoSuchMethodException, InvocationTargetException {
+
+        var beanClass = Object.class;
+        var actual = beanFactory.createBean(beanClass);
+        var expected = beanFactory.getBean(beanClass);
+
+        assertEquals(expected, actual);
+    }
+
+    /**
+     * Test {@link DefaultBeanFactory#createBean(String)} with {@code name}.
+     * <p>
+     * Method under test: {@link DefaultBeanFactory#createBean(String)}
+     */
+    @Test
+    @DisplayName("Test createBean(String) with 'name'")
+    void testCreateBeanWithName() throws NotUniqueBeanDefinitionException {
+        var actual = beanFactory.createBean("Name");
+        var expected = beanFactory.getBean("Name");
+
+        assertEquals(expected, actual);
+    }
+
+    /**
+     * Test {@link DefaultBeanFactory#registerBean(String, BeanDefinition, Object)}.
+     * <p>
+     * Method under test:
+     * {@link DefaultBeanFactory#registerBean(String, BeanDefinition, Object)}
+     */
+    @Test
+    @DisplayName("Test registerBean(String, BeanDefinition, Object)")
+    @Disabled("BeanDefinitionImp is not realized yet")
+    void testRegisterBean() {
+        var beanName = "Name";
```

Comment:
You use the "Name" value as beanName almost in every test case, mb it's better to create  a constant like a TEST_BEAN_NAME = "Name"